marketdata/stream/client.go

```diff
@@ -586,7 +596,18 @@ func (c *client) connReader(
 			return
 		}
 
-		c.in <- msg
+		select {
+		case c.in <- msg:
+		default:
+			c.droppedMsgs++
+			now := time.Now()
+			// Reduce the number of logs to 1 msg/sec if client buffer is full
+			if now.Add(-1 * time.Second).After(c.lastBufferFillMsg) {
+				c.logger.Warnf("datav2stream: writing to buffer failed, error: buffer full, dropped: %d", c.droppedMsgs)
+				c.droppedMsgs = 0
+				c.lastBufferFillMsg = now
+			}
```

Comment:
We should possibly call a handler function instead.

Reply:
Yes, it would be nice. If you do that, please create an option (e.g. `WithBufferFillCallback`), similar to [WithDisconnectCallback](https://github.com/alpacahq/alpaca-trade-api-go/blob/8642da98f386b699a9abca62935630e77e4775e2/marketdata/stream/options.go#L130).